bentoml/yatai/deployment/aws_ec2/operator.py

```diff
@@ -0,0 +1,559 @@
+            )
+        except BentoMLException as error:
+            deployment_pb.state.state = DeploymentState.ERROR
+            deployment_pb.state.error_message = f"Error: {str(error)}"
+            return ApplyDeploymentResponse(
+                status=error.status_proto, deployment=deployment_pb
+            )
+
+    def _update(self, deployment_pb, previous_deployment_pb, bento_path, region):
+        if loader._is_remote_path(bento_path):
+            with loader._resolve_remote_bundle_path(bento_path) as local_path:
+                return self._update(
+                    deployment_pb, previous_deployment_pb, local_path, region
+                )
+
+        updated_deployment_spec = deployment_pb.spec
+        updated_deployment_config = updated_deployment_spec.aws_ec2_operator_config
+
+        describe_result = self.describe(deployment_pb)
+        if describe_result.status.status_code != status_pb2.Status.OK:
+            error_code, error_message = status_pb_to_error_code_and_message(
+                describe_result.status
+            )
+            raise YataiDeploymentException(
+                f"Failed fetching ec2 deployment current status - "
+                f"{error_code}:{error_message}"
+            )
+
+        previous_deployment_state = json.loads(describe_result.state.info_json)
+
+        if "s3_bucket" in previous_deployment_state:
+            s3_bucket_name = previous_deployment_state["s3_bucket"]
+        else:
+            raise BentoMLException(
+                "S3 Bucket is missing in the AWS EC2 deployment, please make sure "
+                "it exists and try again"
+            )
+
+        self.deploy_service(
+            deployment_pb,
+            updated_deployment_spec,
+            bento_path,
+            updated_deployment_config,
+            s3_bucket_name,
+            region,
+        )
+
+        return ApplyDeploymentResponse(status=Status.OK(), deployment=deployment_pb)
+
+    def describe(self, deployment_pb):
+        try:
+            deployment_spec = deployment_pb.spec
+            ec2_deployment_config = deployment_spec.aws_ec2_operator_config
+            ec2_deployment_config.region = (
+                ec2_deployment_config.region or get_default_aws_region()
+            )
+            if not ec2_deployment_config.region:
+                raise InvalidArgument("AWS region is missing")
+
+            deployment_stack_name = generate_aws_compatible_string(
+                "btml-stack-{namespace}-{name}".format(
+                    namespace=deployment_pb.namespace, name=deployment_pb.name
+                )
+            )
+            try:
+                cf_client = boto3.client("cloudformation", ec2_deployment_config.region)
+                cloudformation_stack_result = cf_client.describe_stacks(
+                    StackName=deployment_stack_name
+                )
+                stack_result = cloudformation_stack_result.get("Stacks")[0]
+
```

Comment:
It would be great to get the deployment URL as part of the `info_json`. Sometime, the user might not have the ability to view the address in the web console.

Reply:
This could be bit tricky becase cloudformation api quickly returns without waiting for autoscaling group to spawn instances.
Might need wait -> describe stack polling loop.

Reply:
Yeah,working on it.

Reply:
added instanceDetails in info_json
Is it ok? @yubozhao

Reply:
Yeah, I think that's ok.  And if we can grab out the endpoint URL out, that would be great too, similar to Lambda operator https://github.com/bentoml/BentoML/blob/123fce97a3151b3c467a1105109e793d152ec7a8/bentoml/yatai/deployment/aws_lambda/operator.py#L535-L538

Reply:
updated